src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/BlockApiTest.cs

```diff
@@ -0,0 +1,318 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Core.Modules.Dfs.Tests.Utils;
+using Lib.P2P;
+using MultiFormats;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Modules.Dfs.Tests.IntegrationTests.CoreApi
+{
+    public sealed class BlockApiTest
+    {
+        private IDfsService _dfs;
+        private readonly ITestOutputHelper _output;
+        private const string Id = "QmPv52ekjS75L4JmHpXVeuJ5uX2ecSfSZo88NSyxwA3rAQ";
+        private readonly byte[] _blob = Encoding.UTF8.GetBytes("blorb");
+
+        public BlockApiTest(ITestOutputHelper output)
+        {
+            _output = output;
+            _dfs = TestDfs.GetTestDfs(output, null, "sha2-256");
+        }
+        
+        [Fact]
+        public async Task Put_Bytes()
+        {
+            var cid = await _dfs.BlockApi.PutAsync(_blob);
+            Assert.Equal(Id, cid);
+
+            var data = _dfs.BlockApi.GetAsync(cid).Result;
+            Assert.Equal(_blob.Length, data.Size);
+            Assert.Equal(_blob, data.DataBytes);
+        }
+
+        [Fact]
+        public void Put_Bytes_TooBig()
+        {
+            var data = new byte[_dfs.Options.Block.MaxBlockSize + 1];
+            ExceptionAssert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                var _ = _dfs.BlockApi.PutAsync(data).Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)